# Data Exploration (SMS Spam collection Dataset)

in this notebook,we will:
 - Load the SMS Spam Collection Dataset
 - Check Data Shape
 - View Columns Name
 - Check Basic Info
 - Check Missing Values
 - Check Duplicates
 - Show Unique Labels

This helps us understand the **Pattern of the Datasets** before cleaning or modeling

## Step 1: Set project root for python import

In [1]:
import os
import sys

sys.path.append(os.path.abspath(".."))

## Step 2: Load Dataset

In [2]:
from src.data import (load_data,
data_shape,
col_names,
data_info,
check_mis_val,
check_duplicate,
unique_labels
)
df = load_data(r"D:\Thiru\ML_Projects\GitHub\spam-detection-model\data\raw\spam.csv", 'latin-1')

'Dataset successfully loaded'

v1                                                 v2 Unnamed: 2  \
0   ham  Go until jurong point, crazy.. Available only ...        NaN   
1   ham                      Ok lar... Joking wif u oni...        NaN   
2  spam  Free entry in 2 a wkly comp to win FA Cup fina...        NaN   
3   ham  U dun say so early hor... U c already then say...        NaN   
4   ham  Nah I don't think he goes to usf, he lives aro...        NaN   

  Unnamed: 3 Unnamed: 4  
0        NaN        NaN  
1        NaN        NaN  
2        NaN        NaN  
3        NaN        NaN  
4        NaN        NaN

## Step 3: Check Data Shape

In [3]:
data_shape(df)

data shape(Rows,Columns): (5572, 5)


## Step 4: View Columns names

In [4]:
col_names(df)

columns name: Index(['v1', 'v2', 'Unnamed: 2', 'Unnamed: 3', 'Unnamed: 4'], dtype='object')


## Step 5: Dataset information

In [5]:
data_info(df)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5572 entries, 0 to 5571
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   v1          5572 non-null   object
 1   v2          5572 non-null   object
 2   Unnamed: 2  50 non-null     object
 3   Unnamed: 3  12 non-null     object
 4   Unnamed: 4  6 non-null      object
dtypes: object(5)
memory usage: 217.8+ KB
None


## Step 6: Check Missing Values

In [6]:
check_mis_val(df)

v1               0
v2               0
Unnamed: 2    5522
Unnamed: 3    5560
Unnamed: 4    5566
dtype: int64


## Step 7: check Duplicate

In [7]:
check_duplicate(df)


 Number of duplicate rows:403


## Step 8: Unique Labels

In [8]:
unique_labels(df)


 unique labels in V1 column: ['ham' 'spam']


## Data Exploration Summary
 - Data has `5572` rows and `5` columns
 - Dataset has `3` unnamed columns
 - 3 unnamed columns have more missing values
 - dataset has `403` duplicate rows
 - v1 colunms have `2` unique labels ['ham' 'spam']